headers/VSTPlugin.h

```diff
@@ -19,7 +19,7 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #ifndef OBS_STUDIO_VSTPLUGIN_H
 #define OBS_STUDIO_VSTPLUGIN_H
 
-#define VST_MAX_CHANNELS 8
+#define VST_MAX_CHANNELS 64
```

Comment:
this feels a bit excessive; obs does not handle more than 8 output channels (currently); in the audio pipeline, the vst filter will get audio after it has been resampled to match output sample rate and number of channels, meaning you won't get more than 8.

Reply:
It's true, a proper fix would probably be to use more of the VST api, and properly manage the channel counts for the plugin. The kinds of plugins I am testing with present multi-bus outputs, so up to maybe 32 total channels. OBS will never use these channels without a full re-implementation, but it should at least be able to load use the main bus channels without crashing. This patch is a stop-gap to at least allow them to work.

Reply:
Looks like VST2 plugins do have a fixed numInputs and numOutputs as presented to the host, so the correct thing to do is just pass the right number of buffers into the process callback. The ones past 8 that OBS will never use can just be a temporary dummy buffer. I will re-work the PR.